### PR TITLE
Improve performance when switching between folders

### PR DESCRIPTION
**Pull request by leolost2605**
Getting the folder info always takes a noticeable amount of time as it always gets it from the server if online. That caused switching between folders to have a noticeable delay especially on a slow internet connection. Now we use the folder info that is already available in the `FoldersView` so that we don't have to get it again.

current master:
master
this PR:
improve-performance

### CODE REVIEW

**Comment by zeebok** [comment]
Definitely faster for me as well. Does this change the folders to only update when a refresh occurs, automatically or manually?

**Comment by leolost2605** [comment]
I'm not 100% sure what you mean by "update" (updates about the folder, like name or updates about the messages in the folder?). Also I guess by refresh you mean getting changes about messages? (sry I'm not a native English speaker :))
With this said, this PR only changes the logic around `FolderInfo` in the ConverationList which just holds information about the folder itself (like name, parent, children) and not about any messages. Here it was only used to determine whether it's a "normal" folder or a folder with sent messages. 
The logic around changes affecting messages isn't touched and they will still be reported automatically for all folders at every point of time or a check for messages can still be forced manually with the refresh button. Also once changes are detected the UI still gets updated instantly. 
I hope this might answers your questions? :)

File: src/ConversationList/ConversationList.vala [hunk 1]
```diff
@@ -263,22 +263,22 @@ public class Mail.ConversationList : Gtk.Box {
...
                             }
 
                             try {
                                 var camel_store = (Camel.Store) current_account.service;
 
-                                var folder = yield camel_store.get_folder (current_full_name, 0, GLib.Priority.DEFAULT, cancellable);
+                                var folder = yield camel_store.get_folder (current_folder_info.full_name, 0, GLib.Priority.DEFAULT, cancellable);
                                 folders[current_account.service.uid] = folder;
 
-                                var info_flags = Utils.get_full_folder_info_flags (current_account.service, yield camel_store.get_folder_info (folder.full_name, 0, GLib.Priority.DEFAULT));
```
[the remark this answers is not in the document]
Comment (leolost2605) [reply src/ConversationList/ConversationList.vala:281]: Oh ok if it does change, it gets updated by [these 4 signals](https://github.com/elementary/mail/blob/dd6ddf747bfbeb8bda820f8768ffc2f210bde342/src/FoldersView/AccountSourceItem.vala#L47C28-L47C28)

[the remark this answers is not in the document]
Comment (zeebok) [reply src/ConversationList/ConversationList.vala:281]: Gotcha so this doesn't pertain to any messages only the actual IMAP folders. Okay this makes all a lot more sense then!

**Review by zeebok — APPROVED** [review]
The code seems reasonable and it did make a difference similar to what the gifs show!

**Pull request merged**